src/parser/priest/discipline/modules/shadowlands/covenants/MindGames.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import ItemHealingDone from 'interface/ItemHealingDone';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+import { TooltipElement } from 'common/Tooltip';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+
+import Analyzer from 'parser/core/Analyzer';
+
+import AtonementDamageSource from '../../features/AtonementDamageSource';
+import isAtonement from '../../core/isAtonement';
+
+class MindGames extends Analyzer {
+  static dependencies = {
+    atonementDamageSource: AtonementDamageSource,
+  };
+
+  atonementHealing = 0;
+  directHealing = 0;
+  preventedDamage = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.VENTHYR.id);
+  }
+
+  on_byPlayer_heal(event) {
```

Comment:
For new PRs we much prefer using the new event listener format over the old style. You can see examples of the new style in the Mage or Hunter codebase, but also just as a quick cheat sheet, yours would look like this:

```javascript
  constructor(...args) {
    super(...args);
    this.active = this.selectedCombatant.hasCovenant(COVENANTS.VENTHYR.id);
    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
    this.addEventListener(Events.absorbed.by(SELECTED_PLAYER).spell(SPELLS.MINDGAMES), this.onAbsorb);
  }
  onHeal(event) {
    // same as your on_byPlayer_heal; you technically could probably move the spell check to the event listener but I'm not sure exactly how atonement works
    ....
  }
  // Don't need the spellId check anymore because addEventListener can filter by spell
  onAbsorbed(event) {
    this.preventedDamage += event.amount;
  }
```

Reply:
Fixed